### PR TITLE
For 19886 merge tracking protection shield icon and lock icon.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Amejia481** [comment]
UI tests failing :

- testStrictVisitDisableExceptionToggle
- testStrictVisitProtectionSheet
- testStrictVisitDisable
- testStrictVisitContentShield

**Comment by pocmo** [comment]
There's so much code that it's not easy to digest for me: Is this completely in Fenix or will that be the behavior of `browser-toolbar`? Since we need that in Focus too.

**Comment by Amejia481** [comment]
> There's so much code that it's not easy to digest for me: Is this completely in Fenix or will that be the behavior of `browser-toolbar`? Since we need that in Focus too.

Yeah, I completely understand. As you mentioned all this code is really Fenix specific as we are not really merging both icons together, we are just hide the tracking protection one, and merging all the Fenix UI into  a single unified UI.

**Comment by Amejia481** [comment]
Not sure if it's possible to reuse the UI on AC unless they start looking similar in behavior and look.

**Comment by pocmo** [comment]
> we are just hide the tracking protection one, and merging all the Fenix UI into a single unified UI.

Ah, understood. In that case there's probably not much to share from here. I had the impression there would be a new icon/behavior getting implemented that shows both states somehow. 👍

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt [hunk 1]
```diff
@@ -114,11 +114,11 @@ class DefaultToolbarIntegration(
...
-                if (isPrivate) {
-                    ContextCompat.getColor(context, R.color.primary_text_private_theme)
-                } else {
-                    null
-                }
+            if (isPrivate) {
+                ContextCompat.getColor(context, R.color.primary_text_private_theme)
+            } else {
+                null
+            }
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: Can we check the formatter to make sure this is desired?

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/ConnectionPanelDialogFragment.kt [hunk 2]
```diff
@@ -11,22 +11,26 @@ import android.view.ViewGroup
...
     @VisibleForTesting
-    private lateinit var connectionView: WebsiteInfoView
+    private lateinit var connectionView: ConnectionDetailsView
     private val args by navArgs<ConnectionPanelDialogFragmentArgs>()
+    private var _binding: FragmentConnectionDetailsDialogBinding? = null
 
     override val gravity: Int get() = args.gravity
     override val layoutId: Int = R.layout.fragment_connection_details_dialog
+    // This property is only valid between onCreateView and
+// onDestroyView.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/ConnectionPanelDialogFragment.kt:32]: nit: needs an indent

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: After rebasing I don't see the change

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/ConnectionDetailsView.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    private fun bindUrl(websiteUrl: String) {
+        binding.url.text = websiteUrl
+    }
+
+    private fun bindSecurityInfo(uiValues: WebsiteSecurityUiValues) {
+        val tint = getColor(provideContext(), uiValues.iconTintRes)
+        binding.securityInfo.setText(uiValues.securityInfoRes)
+
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/ConnectionDetailsView.kt:58]: Maybe remove this new line

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt (hunk 1 above)
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: Still seems like it's here?

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt [hunk 4]
```diff
@@ -44,8 +44,10 @@ class QuickSettingsSheetDialogFragment : FenixDialogFragment() {
     private var tryToRequestPermissions: Boolean = false
     private val args by navArgs<QuickSettingsSheetDialogFragmentArgs>()
 
-    private lateinit var binding: FragmentQuickSettingsDialogSheetBinding
-
+    private var _binding: FragmentQuickSettingsDialogSheetBinding? = null
+    // This property is only valid between onCreateView and
+// onDestroyView.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt:49]: nit: needs an indent

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: When you directly to files that have changed on the PR https://github.com/mozilla-mobile/fenix/pull/20536/files, I'm not able to see the change, are you?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: I moved the line above as ktlint was complaining

**Pull request merged**